### DISCUSSION TITLE
Could I use GNNExplainer to explain the result of edge regression?

### DISCUSSION THREAD

**Discussion by LuoMiao77** [pyg-team/pytorch_geometric#4547] [Q&A]
Hi, I am trying to use GNNExplainer to explain the result of edge regression, but it said that the shape of the mask [24758] at index 0 (the number of edges)does not match the shape of the indexed tensor [366] at index 0(the number of nodes) .

**Comment by rusty1s** [comment]
Do you have a small example to reproduce?

**Reply by LuoMiao77** [reply]
I set up the folowing model to predict the flow between 2 nodes, and I want to explain the result of edge regression by explaining the contributions of the features of nodes.
```python
class Model(nn.Module):
    def __init__(self,in_feats,n_xidden,out_classes,dropout=0.5):
        super().__init__()   
        self.dropout=nn.Dropout(p=dropout)
        self.conv1=GCNConv(in_channels=in_feats,out_channels=n_xidden)
        self.conv2=GCNConv(in_channels=n_xidden,out_channels=n_xidden)
        self.conv3=GCNConv(in_channels=n_xidden,out_channels=out_classes)
        self.bn1 = nn.BatchNorm1d(num_features=n_xidden) 
        self.bn2 = nn.BatchNorm1d(num_features=n_xidden)
    def forward(self,x, edge_index, edge_weight):      
        x=self.conv1(x,edge_index,edge_weight)
        x=self.bn1(x)
        x=F.relu(x)
        x=self.dropout(x)
        x=self.conv2(x,edge_index,edge_weight)
        x = self.bn2(x)
        x = F.relu(x)
        x =self.dropout(x)
        x = self.conv3(x,edge_index,edge_weight)
        src, dst = edge_index
        score = (x[src] * x[dst]*edge_weight).sum(dim=-1)
        return score
```

**Reply by rusty1s** [reply]
Ah, thanks for clarifying. I misread. `GNNExplainer` does not support explaining edges (only nodes and graphs). You may want to use `Captum` for this, see https://github.com/pyg-team/pytorch_geometric/discussions/4058.
